### DISCUSSION TITLE
Documentation update poposal: How to upgrade from django-q to django-q2

### DISCUSSION THREAD

**Discussion by klausfrandsen** [django-q2/django-q2#97] [Ideas]
So happy to find that django-q lives on in django-q2. However, I am not able to figure out how much work is involved in "migrating" my django app over from django-q to django-q2. 

More specific questions that comes to mind:
What amount of code changes are required
Can tasks be migrated from q to q2 (given that the ORM is the broker)

I'm looking into upgrading from q to q2, and maybe I can contribute my learnings to the docs.

**Comment by GDay** [comment]
> What amount of code changes are required

No code changes are required

>  Can tasks be migrated from q to q2 (given that the ORM is the broker)

I don't recall any changes that should make this fail. So It should work fine.

Django-q and django-q2 are not _that_ much different. I have made sure to keep them compatible so that migration is as easy as removing the old package, installing the new package and then running migrations. That's it, really.

It might be a good idea to look through the changelog just to see if any of the new features (like "alternative clusters" and "intended date")  appeal to you.

**Comment by msabatier** [comment]
For Schedules I confirm that you can migrate from django q to q2 without any problem. You only need to run the database migrations, nothing else is needed.
We did such upgrade last week without issue. 
Unfortunately we did not have any queued task at the time of migration, so I cannot comment on that.

**Comment by mast3rbow** [comment]
One area to be aware of is, 

If you are using some Django_q Addons such as Django_q_tenants etc, you will need to fork them and update their dependencies to leverage DjangoQ2. Just something worth noting that I had to resolve.
